File: programmers/level2/carpet.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(int brown, int yellow) {
    vector<int> answer;
    int area = brown + yellow;
    for(int height = 3; ; height++) {
        if(!(area % height)) {
            int width = area / height;
            if((width - 2) * (height - 2) == yellow) {
                answer.push_back(width);
                answer.push_back(height);
                break;
            }
        }
    }
    return answer;
}

File: programmers/level2/Carpet.java
import java.util.*;

class Solution {
    public int[] solution(int brown, int yellow) {
        int[] answer = {};

        int maxValue = (int)Math.sqrt(brown + yellow);
        int a = 0;
        for(int b = 3; b <= maxValue; b++) {
            if((brown + 4) % 2 != 0) {
                continue;
            }
            a = (brown + 4 - (2 * b)) / 2;
            if(a * b != brown + yellow) {
                continue;
            }
            answer = new int[]{a, b};
            break;
        }
        return answer;
    }
}
